sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/ResourceBuilder.java

```diff
@@ -190,8 +190,9 @@ public ResourceBuilder setMesosResource(MesosResource mesosResource) {
     public Resource build() {
         Resource.Builder builder =
                 mesosResource == null ? Resource.newBuilder() : mesosResource.getResource().toBuilder();
-        builder.setName(resourceName).setRole(preReservedRole);
-        builder.setType(value.getType());
+        builder.setName(resourceName)
```

Comment:
Changes here and below are the bug fix, I needed to make to resource refinement work again on `testing/pull/1524`